### PR TITLE
Feedback submission errors

### PR DESCRIPTION
**Pull request by lihanli**
## Description of change

fix gi bill feedback submission errors

## Testing done
tested on staging

## Testing planned
test on staging/dev on website

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: app/models/gi_bill_feedback.rb [hunk 1]
```diff
@@ -69,8 +71,12 @@ def transform_form
 
     transformed['issue'] = transform_keys_into_array(transformed['issue'])
     transformed['email'] = transformed.delete('anonymous_email') || transformed.delete('applicant_email')
+    transformed = Common::HashHelpers.deep_compact(transformed)
+    transformed.delete('profile_data') if transformed['profile_data'].blank?
```
Comment (jholton) [inline app/models/gi_bill_feedback.rb]: We're adding `profile_data` and then deleting it if it's blank.
If you rollback my changes from [here](https://github.com/department-of-veterans-affairs/vets-api/pull/2264/files), then `profile_data` is not set at all if `user.blank?` and then you don't need this line.

Comment (lihanli) [reply app/models/gi_bill_feedback.rb]: the last four ssn line assumes profile data hash exists and it will be set even if `user.blank?`, also in the tests it will sometimes set `profile_data` to `{ 'active_ICN' => nil }` which then gets compacted to `{}` by `deep_compact` and will still need to be deleted

Comment (jholton) [reply app/models/gi_bill_feedback.rb]: Makes sense.
